gqcpy/src/QCMethod/HF/UHF/UHFSCFEnvironment_bindings.cpp

```diff
@@ -56,28 +56,26 @@ void bindUHFSCFEnvironment(py::module& module) {
 
         .def_static(
             "WithCoreGuess",
-            [](const size_t N_alpha, const size_t N_beta, const USQHamiltonian<double>& sq_hamiltonian, const Eigen::MatrixXd& S) {  // use an itermediary Eigen matrix for the Python binding, since Pybind11 doesn't accept our types that are derived from Eigen::Matrix
-                return UHFSCFEnvironment<double>::WithCoreGuess(N_alpha, N_beta, sq_hamiltonian, SquareMatrix<double>(S));
+            [](const size_t N_alpha, const size_t N_beta, const USQHamiltonian<double>& sq_hamiltonian, const ScalarUSQOneElectronOperator<double>& S) {
+                return UHFSCFEnvironment<double>::WithCoreGuess(N_alpha, N_beta, sq_hamiltonian, S);
             },
             "Initialize an UHF SCF environment with initial coefficient matrices (equal for alpha and beta) that is obtained by diagonalizing the core Hamiltonian matrix.")
 
 
         // Bind read-write members/properties, exposing intermediary environment variables to the Python interface.
-        .def_readwrite("N_alpha", &UHFSCFEnvironment<double>::N_alpha)
-        .def_readwrite("N_beta", &UHFSCFEnvironment<double>::N_beta)
+        .def_readwrite("N", &UHFSCFEnvironment<double>::N)
 
         .def_readwrite("electronic_energies", &UHFSCFEnvironment<double>::electronic_energies)
 
-        .def_readwrite("orbital_energies_alpha", &UHFSCFEnvironment<double>::orbital_energies_alpha)
-        .def_readwrite("orbital_energies_beta", &UHFSCFEnvironment<double>::orbital_energies_beta)
+        .def_readwrite("orbital_energies", &UHFSCFEnvironment<double>::orbital_energies)
```

Comment:
Does this also require a Python binding for `SpinResolved<...>`?

Reply:
It does, but I believe `SpinResolved<VectorX<double>>` already has a binding. I could be mistaken and I'll be sure to check before merging this PR!

Reply:
Ok, I didn't remember.